### PR TITLE
[receiver/collectd] remove opencensus

### LINKED ISSUE #25148: [receiver/collectd] Refactor Observability.go from opencensus to pdata
### Component(s)

receiver/collectd

### Is your feature request related to a problem? Please describe.

Hi,
I hope someone can refactor [collectdreceiver/observability.go](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/collectdreceiver/observability.go#L10). 
It appears that certain sections of the code still rely on opencensus.

```
import (
	"context"

	"go.opencensus.io/stats"
	"go.opencensus.io/stats/view"
)
```

Furthermore, I've noticed that there are other collectors that also employ similar logic for handling `observability.go` (ex: [k8sattributesprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/internal/observability/observability.go)).

### Describe the solution you'd like

Hope people can reactor part of observability.go to pdata.

### Describe alternatives you've considered

_No response_

### Additional context

From what I know, the purpose of this code is to record events of success and failure. 

I've seen someone use module [obsreport](https://pkg.go.dev/go.opentelemetry.io/collector/obsreport)  in opentelemetry-collector-contrib to achieve similar functionality, For example, in the [splunkhecreceiver](https://github.com/ZenoCC-Peng/opentelemetry-collector-contrib/blob/main/receiver/splunkhecreceiver/receiver.go#L233).

However, the metric mode in obsreport refers to "[go.opentelemetry.io/otel/metric](http://go.opentelemetry.io/otel/metric),"  which refers from https://pkg.go.dev/go.opentelemetry.io/otel/metric and it's different from "pmetric".

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
use telemetry settings, don't rely on opencensus for component metrics

**Link to tracking Issue:**
Fixes #25148

### CODE REVIEW

**Review by djaglowski — APPROVED** [review]
LGTM

**Review by songy23 — APPROVED** [review]

File: receiver/collectdreceiver/observability.go [hunk 1]
```diff
@@ -6,79 +6,66 @@ package collectdreceiver // import "github.com/open-telemetry/opentelemetry-coll
...
-}
+func initMetrics(telemetrySettings component.TelemetrySettings) (*observability, error) {
+	meter := telemetrySettings.MeterProvider.Meter("collectd")
 
-var viewBlankDefaultAttrs = &view.View{
-	Name:        mBlankDefaultAttrs.Name(),
-	Description: mBlankDefaultAttrs.Description(),
-	Measure:     mBlankDefaultAttrs,
-	Aggregation: view.Sum(),
+	mErrors, err := meter.Int64Counter("errors", metric.WithUnit("1"), metric.WithDescription("Errors encountered during processing of collectd requests"))
```
Comment (codeboten) [inline receiver/collectdreceiver/observability.go:24]: is the metric name preserved with this change?

Comment (atoulme) [reply receiver/collectdreceiver/observability.go:24]: No, in all likelihood I broke this.

**Review by codeboten — CHANGES_REQUESTED** [review]
Requesting change to ensure the metric naming isn't broken. If the metric naming changes, the breaking change procedure should be followed :)

File: receiver/collectdreceiver/observability.go (hunk 1 above)
Comment (atoulme) [reply receiver/collectdreceiver/observability.go:24]: I meant to change those names anyway. I don't think "otel/collectd/errors" was a good name. I'm going to look for ways to also make those metrics optional by default. So we're going for a breaking change here.

**Comment by atoulme** [comment]
Breaking it all the way - no more metrics, let's use obsrecv instead.

**Review by codeboten — APPROVED** [review]
I think this is ok since the component is still marked as alpha, for beta components i would expect this change would require a feature gate 👍🏻

**Comment by atoulme** [comment]
Indeed. Note I have opened an issue to move this component to beta here: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28658 and am doing this change now because of it, partially.

**Pull request merged**